eng/common/pipelines/templates/stages/archetype-typespec-emitter.yml

```
@@ -0,0 +1,452 @@
+#   file: The path to the file to publish. This is relative to the packages directory in the build artifacts directory.
+- name: Packages
+  type: object
+  default: []
+
+# Whether to publish to the internal feed.
+- name: PublishInternal
+  type: boolean
+  default: true
+
+# Indicates if the Publish stage should depend on the Test stage
+- name: PublishDependsOnTest
+  type: boolean
+  default: false
+
+# Whether to publish to the internal feed.
+- name: ShouldRegenrate
+  type: boolean
+  default: false
+
+# Number of jobs to generate. This is the maximum number of jobs that will be generated. The actual number of jobs will be reduced if it would result in fewer than MinimumPerJob packages per job.
+- name: RegenerationJobCount
+  type: number
+  default: 10
+
+# Minimum number of packages to generate per job.
+- name: MinimumPerJob
+  type: number
+  default: 10
+
+# Indicates if regenration matrix should only contain folders with typespec files
+- name: OnlyGenerateTypespec
+  type: boolean
+  default: false
+
+stages:
+
+# Build stage
+# Responsible for building the autorest generator and typespec emitter packages
+# Produces the artifact `build_artifacts` which contains the following:
+#   package-versions.json: Contains a map of package name to version for the packages that were built
+#   overrides.json: Contains npm package version overrides for the emitter and generator
+#   packages/: Contains the packages to publish
+#   lock-files/: Contains package.json and package-lock.json files for use in the test stage to ensure tests are run against the correct dependency versions
+- stage: Build
+  pool: ${{ parameters.Pool }}
+  jobs:
+  - job: Build
+    steps:
+    # Validate parameters and fail early if invalid
+    - ${{ each package in parameters.Packages }}:
+      - ${{ if notIn(package.type, 'npm', 'nuget') }}:
+        - script: |
+            echo "Package ${{ package.name }} has unsupported type: ${{ package.type }}"
```

I think 1es pt figured out how to surface an error message at the pipeline validation level, possibly with something like https://dev.azure.com/azure-sdk/_git/1ESPipelineTemplates?path=/v1/Stages/ReleaseStage.yml&version=GBmain&line=31&lineEnd=34&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

However I guess a downside of that approach is we will omit a check suite to github, so it can be a silent failure when expected as a triggered pipeline.

---

I like the `:error` syntax, but ended up removing this whole check when I removed the `Packages` parameter